clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -363,6 +365,20 @@ private CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> chainFuture(fi
                 }
             });
         }
+
+        @Override
+        public String toString() {
+            return "OffsetFetchRequestState{" +
+                    "requestedPartitions=" + requestedPartitions +
+                    ", requestedGeneration=" + requestedGeneration +
+                    ", future=" + future +
+                    ", exponentialBackoff=" + exponentialBackoff +
+                    ", lastSentMs=" + lastSentMs +
+                    ", lastReceivedMs=" + lastReceivedMs +
+                    ", numAttempts=" + numAttempts +
+                    ", backoffMs=" + backoffMs +
```

Comment:
Instead of duplicating the code in RequestState, could we pull out the common part as a util and reuse?

Reply:
@junrao Are you referring to the `toString()` code specifically, or something else?

Reply:
I can make an internal `toStringBase()` to `RequestState` that has the core instance variables concatenated as a string. And then subclasses can first append their own instance variable string and then append the result of `toStringBase()` or something. Does that make sense?

Reply:
Yes, I was referring to the `toString` method. It seems that every subclass of `RequestState` directly gets every field in `RequestState` to build its own string. This creates duplicated code and can be a bit error prone.

Reply:
@junrao I created `toStringBase()` in `RequestState`:

```java
/**
 * This method appends the instance variables together in a simple String of comma-separated key value pairs.
 * This allows subclasses to include these values and not have to duplicate each variable, helping to prevent
 * any variables from being omitted when new ones are added.
 *
 * @return String version of instance variables.
 */
protected String toStringBase() {
    return "owner='" + owner + '\'' +
            ", exponentialBackoff=" + exponentialBackoff +
            ", lastSentMs=" + lastSentMs +
            ", lastReceivedMs=" + lastReceivedMs +
            ", numAttempts=" + numAttempts +
            ", backoffMs=" + backoffMs;
}

@Override
public String toString() {
    return "RequestState{" + toStringBase() + '}';
}
```

Subclasses look like this:

```java
@Override
public String toString() {
    return "OffsetFetchRequestState{" +
            "requestedPartitions=" + requestedPartitions +
            ", requestedGeneration=" + requestedGeneration +
            ", future=" + future +
            ", " + toStringBase() +
            '}';
}
```

Does that seem OK?